pkg/archive/archive_windows.go

```diff
@@ -6,9 +6,25 @@ import (
 	"archive/tar"
 	"fmt"
 	"os"
+	"path/filepath"
 	"strings"
 )
 
+// fixVolumePathPrefix does platform specific processing to ensure that if
+// the path being passed in is not in a volume path format, convert it to one.
+func fixVolumePathPrefix(srcPath string) string {
+	if !strings.HasPrefix(srcPath, `\\?\`) {
+		srcPath = `\\?\` + srcPath
+	}
+	return srcPath
+}
+
+// getWalkRoot calculates the root path when performing a TarWithOptions.
+// We use a seperate function as this is platform specific.
+func getWalkRoot(srcPath string, include string) string {
+	return filepath.Join(srcPath, include)
```

Comment:
I understand the rationale for not using `filepath.Join` on Linux, but what's wrong about using the same string concatenation on Windows too? That is: using `getWalkRoot` as it is implemented for Linux on all platforms.